### PR TITLE
CI: Add CODEOWNERS file with math-fehr ownership

### PR DESCRIPTION
**Pull request by math-fehr**
This will set defaults reviewers when a new PR is added.

### CODE REVIEW

**Review by webmiche — APPROVED** [review]
LGTM, you can also throw me in there, in particular the CI might make sense.

**Comment by webmiche** [comment]
Okay, I did something. I think we should briefly discuss how we want to do this. AFAIU, the file is scanned from bottom up and the first match is the people that are requested. So, IMO, we should have a couple of people as the defaults for most folders and then overwrite that default by making specific files owned by others.

For example, I went ahead and added me and @math-fehr  as the default for the dialects, as we are interested in most of them. I now imagine the RISCV people will now take over the riscv file in that folder. Does this make sense? :sweat_smile:

**Comment by math-fehr** [comment]
Yes, that's the plan at least!
I put myself as default, and overwrote some places that I want to take care of.
If you want to take care of a dialect specifically, you should put yourself in the down of the list.

**Comment by compor** [comment]
I added people for the RISCV dialects (not complete I assume, those with the easiest gh handles)

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+xdsl/dialects/*               @math-fehr @webmiche
+
+# xDSL RISCV dialects
+xdsl/dialects/riscv*          @AntonLydike @superlopuh @compor @nazavode
+
+# xDSL frontend
+xdsl/frontend/**              @webmiche
+
+# xDSL documentation
+docs/**                       @math-fehr @webmiche
```
Comment (superlopuh) [inline .github/CODEOWNERS:28]: ✋

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+# xDSL benchmarks
+bench/**                      @math-fehr
```
Comment (superlopuh) [inline .github/CODEOWNERS:31]: ✋

**Review by superlopuh — APPROVED** [review]

**Comment by superlopuh** [comment]
It's worth noting that this is slightly different from notiffications. I have notifications set up for all pull requrests and issues in xdsl, and we might want to use the owners file as a way to mark areas of code where the owner needs to approve to merge. I've been adding people as reviewers to notify them that I've opened a PR, but maybe we'd be better off with CC and @ comments in the PR itself. Or we could stick with what we've been doing :)

File: .github/CODEOWNERS (hunk 1 above)
Comment (superlopuh) [inline .github/CODEOWNERS:28]:
```suggestion
docs/**                       @math-fehr @webmiche @superlopuh 
```

File: .github/CODEOWNERS (hunk 2 above)
Comment (superlopuh) [inline .github/CODEOWNERS:31]:
```suggestion
bench/**                      @math-fehr @superlopuh 
```

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+# These owners will be the default owners for everything in
+# the repo. Unless a later match takes precedence,
+*                             @math-fehr @webmiche
+
+
+
+# xDSL core
+xdsl/*                        @math-fehr @webmiche
+xdsl/utils/**                 @math-fehr
```
Comment (superlopuh) [inline .github/CODEOWNERS:13]:
```suggestion
xdsl/utils/**                 @math-fehr @superlopuh
```

**Comment by AntonLydike** [comment]
I just did the HPC stuff, and I think we should agree on how we handle experimental. Ideally we set experimental up to have no code owners? Any opinions on that?

**Review by AntonLydike — APPROVED** [review]

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+# xdsl-opt tool
+xdsl/tools/**                 @math-fehr @webmiche
+
+# xDSL dialects
+xdsl/dialects/*               @math-fehr @webmiche
+
+# xDSL RISCV dialects
+xdsl/dialects/riscv*          @AntonLydike @superlopuh @compor @nazavode
+
```
Comment (nazavode) [inline .github/CODEOWNERS:23]:
```suggestion

# xDSL RISCV extension dialects
xdsl/dialects/snitch.py       @nazavode
xdsl/transforms/*snitch*      @nazavode

```

Adding myself for the currently available RISC-V extension dialects in xDSL along the lines of the MPI stuff @AntonLydike

**Review by meshtag — APPROVED** [review]
I added myself for stencil and vector as they seem most interesting to me atm, and I have done some work there.

**Review by nazavode — APPROVED** [review]

**Pull request merged**